clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -153,7 +154,16 @@ void runOnce() {
                 .map(rm -> rm.maximumTimeToWait(currentTimeMs))
                 .reduce(Long.MAX_VALUE, Math::min);
 
-        reapExpiredApplicationEvents(currentTimeMs);
+        List<CompletableEvent<?>> completableEvents = reapExpiredApplicationEvents(currentTimeMs);
+        
+        // If there is a metadata error, complete the completable events which are not passed by error event with the metadata error
+        if (networkClientDelegate.metadataError().isPresent()) {
+            Throwable metadataError = networkClientDelegate.metadataError().get();
+            completableEvents.stream()
+                    .filter(event -> !(event instanceof CompletableApplicationEvent && ((CompletableApplicationEvent<?>) event).isPassedByErrorEvent()))
+                    .forEach(event -> event.future().completeExceptionally(metadataError));
+            networkClientDelegate.clearMetadataError();
```

Comment:
if there were no events to propagate the error, this line will still clear the error. Shouldn't we clear only if we were able to propagate it?